Extension/src/packageManager.ts

```diff
@@ -300,7 +317,14 @@ export class PackageManager {
                             }
                         });
 
-                        response.on('end', resolve);
+                        response.on('end', () => {
+                            const packageBuffer: Buffer = Buffer.concat(buffers);
+                            if (isValidPackage(packageBuffer, pkg.integrity)) {
+                                resolve();
+                            } else {
+                                reject(new PackageManagerError('Invalid content received. Hash is incorrect.', localize("invalid.content.received", 'Invalid content received. Hash is incorrect.'), 'DownloadFile', pkg));
```

Comment:
Although this whole scenario is unlikely, should this error be more geared at advising the user what to do about the issue?

Reply:
If the hash incorrect, then the whole install would fail. I'll see what messages is further given to the user when the whole install is incomplete or if the entire install is undone.

Reply:
I tested that if it rejects the downloaded package then, the install fails. Failed installs results in message to reinstall again.